src/encoder.rs

```diff
@@ -1770,7 +1778,9 @@ fn encode_partition_topdown(seq: &Sequence, fi: &FrameInvariants, fs: &mut Frame
 
     match partition {
         PartitionType::PARTITION_NONE => {
-            let part_decision = if !rdo_output.part_modes.is_empty() {
+            // FIXME: For now always redo RDO since prior RDO may have relied on inaccurate
+            // neighbours
```

Comment:
I am trying to understand which case redoing rdo_mode_decision()  is required.
I assume you have tested locally this patch stops the corruption.

Reply:
The case is where you have a first split. Then RDO is done on 4 smaller blocks. The best parameters for the last 3 smaller blocks assume that the previous smaller blocks were not split. But these blocks may actually end up being split. In such case the NEARESTMV may not be the same anymore, which causes the issue.

Reply:
I think, in such case, even intra mode(s) of the 1st smaller block can change if it has split?